### PR TITLE
Parameterize database config to support docker-compose

### PR DESCRIPTION
**Pull request by tjwp**
This parameterization is the same that is used in our other docker-composed applications.

In the `docker-compose.yml` file I've also removed the obsolete, unnecessary `links` and added `depends_on`.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,8 +19,6 @@ services:
     volumes:
       - .:/usr/src/myapp
       - bundle-volume:/usr/src/bundler
-    links:
```
Comment (zoso10) [inline docker-compose.yml]: nice, meant to remove this since i saw it's become a legacy feature

File: config/database.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@
 default: &default
   adapter: postgresql
   encoding: unicode
+  host: <%= ENV.fetch("POSTGRES_HOST") { "localhost" } %>
```
Comment (zoso10) [inline config/database.yml]: how are we managing the env vars being set?

Comment (zoso10) [reply config/database.yml]: :upside_down_face: lol looks like we already have them set in our docker-compose.yml

Comment (tjwp) [reply config/database.yml]: They are set in the `docker-compose.yml`, or the defaults here are used.

**Pull request merged**